tests/workflows/test_cluster_plots.py

```diff
@@ -0,0 +1,49 @@
+import json
+import os
+
+import luigi
+import pytest
+from google.cloud import storage
+
+from workflows.cluster_plots.__main__ import ClusterPlotTaskWrapper
+
+default_output_path = "/home/nzhon/data/processed/birdclef-2022/birdnet-embeddings-with-neighbors-static/v1"
+
+
+def test_cluster_plots_task_luigi_build(output_path=default_output_path):
+    f = open(
+        "/home/nzhon/data/processed/birdclef-2022/birdnet-embeddings-with-neighbors-agreement-static/v1/agreement.json"
+    )
+    agreement = json.load(f)
+    species_list = [t["ego_primary_label"] for t in agreement]
+
+    task = ClusterPlotTaskWrapper(
+        output_path=output_path,
+        list_species=species_list,
+    )
+    res = luigi.build([task], local_scheduler=True, workers=1, detailed_summary=True)
+    assert res
+    output_dir = list(os.listdir(output_path))
+    assert len(output_dir) == 23
+    # assert(output_dir == len(list_species))
+
+    storage_client = storage.Client("birdclef-2023")
```

Comment:
Hrmm, so we don't actually want to run the GCS part of the code when we do unit testing. I wonder if we could mock out the functionality here, either with [monkeypatch](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html) (which iirc I didn't have any luck here), or to to pass a parameter that marks the downstream pull/push task to be run in a mocked mode for testing.

Reply:
for now i took the gcs part out of the unit test. my thought process is that since they are independent tasks, we can write unit tests for push/pull separately instead of testing it with my workflow

Reply:
Yep, that's a fair approach. We do want to make sure that the GCS stuff doesn't accidentally get run as part of the tests though, or at least written to a scratch location in the bucket.